spec/src/main/asciidoc/ch03-entity-operations.adoc

```diff
@@ -6122,6 +6122,30 @@ _?1_ ”).
 
 The use of positional parameters is not supported for criteria queries.
 
+==== Arguments to query parameters
+
+Arguments are assigned to query parameters by calling `Query.setParameter()`.
+The first parameter of `setParameter()` identifies the named or positional
+parameter of the query.
+
+An argument may be assigned to a single-valued parameter of a JPQL or
+native SQL query by passing the argument to the second parameter of
+`setParameter()`.
+
+[source,java]
+----
+query.setParameter("name", name)
+----
+
+A list of arguments may be assigned to a collection-valued parameter of
+a JPQL query by passing the arguments in an instance of `java.util.List`
```

Comment:
Shouldn't it rather be Collection?

Reply:
I dunno, should it?

I would have thought `List` was more natural here. It's a parameter list, after all.

Reply:
It's named `collection-valued parameter` after all. Also, in Hibernate we support collection, not just list. I don't know why it is natural to expect the in-predicate parameter **collection** to be ordered and hence a `List`. I would rather argue it's more natural to expect a `Set`, because duplicate values don't make sense, but either way, let's stick to `Collection`.

Reply:
>  I don't know why it is natural to expect the in-predicate parameter collection to be ordered and hence a List. 

Well because it looks (syntactically) like a tuple constructor.

And so if we ever wanted to generalize this to other things were tuple constructors make sense, we would want order to be significant. e.g. `insert into Book values :properties` or `where (a,b,c) = :tuple`, etc.

I know we don't support these things in JPA _now_, so it's not a strong argument, but we actually _do_ have tuple constructors and `insert ... values` in Hibernate, so it's not crazy to think that we might add them to JPQL someday.